cmd/policies.go

```diff
@@ -309,3 +325,345 @@ func viewPolicyCmd(ctx *cli.Context) error {
+			return
+		}
+
+		if teams == nil || len(teams) == 0 {
+			teamErr = errs.NewExitError("No teams found for organisation.")
+			return
+		}
+	}()
+	wg.Wait()
+
+	if userErr != nil || teamErr != nil {
+		return nil, cli.NewMultiError(userErr, teamErr,
+			errs.NewExitError(policyTestFailed),
+		)
+	}
+
+	teamChan := make(chan envelope.Team, 1)
+	filterTeamsByUser(client, teams, user, teamChan, orgID)
+
+	result := make([]envelope.Team, 0)
+	for t := range teamChan {
+		result = append(result, t)
+	}
+	return result, nil
+}
+
+// Given a set of teams and a user, filter (exclude) the teams for which the
+// user is not a member. This requires getting the membership information for
+// each team, which can be expensive, but can be done in parallel.
+func filterTeamsByUser(client *api.Client, teams []envelope.Team,
+	user *apitypes.Profile, teamChan chan envelope.Team,
+	orgID *identity.ID) {
+
+	c := context.Background()
+	wg := &sync.WaitGroup{}
+	go func() {
+		for _, t := range teams {
+			if isMachineTeam(t.Body) {
+				continue
+			}
+
+			wg.Add(1)
+			go func(team envelope.Team) {
+				defer wg.Done()
+				members, err := client.Memberships.List(c, orgID, team.ID, nil)
+				if err != nil {
+					// TODO: Log this!!!
+					return
+				}
+				for _, m := range members {
+					if *user.ID == *m.Body.OwnerID {
+						teamChan <- team
+						return
+					}
+				}
+			}(t)
+		}
+		wg.Wait()
+		close(teamChan)
+	}()
+}
+
+// Fetch all policies and policy-attachments for the org. The two steps are
+// independent so do them in parallel.
+func getPoliciesAndAttachments(client *api.Client, orgID *identity.ID) ([]envelope.Policy, []envelope.PolicyAttachment, error) {
+	c := context.Background()
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	var pErr error
+	var policies []envelope.Policy
+	go func() {
+		defer wg.Done()
+		policies, pErr = client.Policies.List(c, orgID, "")
+	}()
+
+	var aErr error
+	var attachments []envelope.PolicyAttachment
+	go func() {
+		defer wg.Done()
+		attachments, aErr = client.Policies.AttachmentsList(c, orgID, nil, nil)
+	}()
+	wg.Wait()
+
+	if aErr != nil || pErr != nil {
+		return nil, nil, cli.NewMultiError(pErr, aErr,
+			errs.NewExitError(policyTestFailed))
+	}
+	return policies, attachments, nil
+}
+
+// Test if a Policy satisfies some condition
+type PolicyPredicate func(envelope.Policy) bool
+
+// Compound Predicate: Test of a policy satisfies all predicates
+func AllPredicate(predicates ...PolicyPredicate) PolicyPredicate {
+	return func(policy envelope.Policy) bool {
+		for _, pred := range predicates {
+			if !pred(policy) {
+				return false
+			}
+		}
+		return true
+	}
+}
+
+func filterPolicies(policies []envelope.Policy, predicate PolicyPredicate) []envelope.Policy {
+	result := make([]envelope.Policy, 0)
+	for _, pol := range policies {
+		if predicate(pol) {
+			result = append(result, pol)
+		}
+	}
+	return result
+}
+
+// Crete a Predicate to filter (exclude) policies that are not attached to any
```

Comment:
godoc comments always start with the func name, like:
` policyAttachedToTeamsPredicate creates a Predicate to...`